### PR TITLE
Add playwright e2e testing

### PR DESCRIPTION
**Pull request by sirius651**
**Pull Request Summary**

Add playwright e2e testing

**Check list**

- [ ] contains breaking changes
- [x] adds new feature
- [ ] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**Release notes:**

add: playwright testing workflow

### CODE REVIEW

**Comment by sirius651** [comment]
I completed brief example of test cases in dapp-staking page [here](https://github.com/AstarNetwork/astar-apps/blob/add/playwright/tests/dappstaking.spec.ts) , I also want to discuss whether playwright helps while you try to writing e2e testing for the [assets](https://github.com/AstarNetwork/astar-apps/blob/add/playwright/tests/assets.spec.ts) page or the [dashboard](https://github.com/AstarNetwork/astar-apps/blob/add/playwright/tests/dashboard.spec.ts) page. @impelcrypto @bobo-k2

**Review by bobo-k2 — APPROVED** [review]
LGTM

File: tests-examples/demo-todo-app.spec.ts [hunk 1]
```diff
@@ -0,0 +1,425 @@
+//@ts-ignore
+import { test, expect, Page } from '@playwright/test';
```
[the remark this answers is not in the document]
Comment (sirius651) [reply tests-examples/demo-todo-app.spec.ts]: we can remove it. removed now

**Comment by sirius651** [comment]
> Hi @sirius651 I got this error. Do you know what might be issue? I've installed both dependencies and playwright.
> 
it seems error says `Protocol error - cannot navigate to invalid URL` . unfortunately, I couldn't reproduce like that. can you try like this as put absolute path?
```
test.beforeEach(async ({ page }) => {
await page.goto('http://localhost:8080/astar/dapp-staking/discover');
})
```

**Comment by impelcrypto** [comment]
@sirius651 Sorry, it looks like the double quotation (`""`) I used in https://github.com/AstarNetwork/astar-apps/pull/703#issuecomment-1495912963 was wrong. I've modified and it worked!

**Review by impelcrypto — APPROVED** [review]

**Review by niklabh — APPROVED** [review]

**Comment by sirius651** [comment]
merge this PR now, and once the rpc environment is ready, we'll create another PR separately

**Pull request merged**